### PR TITLE
Adding lease schema for ledger compatibility

### PR DESCRIPTION
**Pull request by brunodb3**
- Adds `leaseSchemaLedger` for compatibility
- Renames `transferSchemaV1` to `transferSchemaLedger`, for more clarity
- Once ledger app is fixed, these schemas are no longer needed
  - https://github.com/iicc1/ledger-app-lto/issues/3

### CODE REVIEW

File: src/parseSerialize/schemas.ts [hunk 1]
```diff
@@ -329,15 +329,11 @@ export const setScriptSchemaV1: TSchema = {
   ],
 }
 
-// This schema is not 100% correct for transfer-v1, it's a workaround for an issue with the Ledger app in the wallet ui
-// Once ledger app is fixed, this will no longer be needed
-// see the issue: https://github.com/iicc1/ledger-app-lto/issues/3
-export const transferSchemaV1: TSchema = {
+export const transferSchemaV2: TSchema = {
   type: 'object',
   schema: [
```
Comment (KaloyanTanev) [inline src/parseSerialize/schemas.ts:334]: There should be attachment length field before the attachment:
https://docs.ltonetwork.com/v/edge/protocol/public/transactions/transfer-transaction

File: src/parseSerialize/schemas.ts [hunk 2]
```diff
@@ -347,21 +343,20 @@ export const transferSchemaV1: TSchema = {
   ],
 }
 
-export const transferSchemaV2: TSchema = {
+export const sponsorSchemaV1: TSchema = {
   type: 'object',
   schema: [
```
Comment (KaloyanTanev) [inline src/parseSerialize/schemas.ts:348]: Not sure if the order matters (if you use this order for actually producing the signature), if it does, timestamp and fee should be the other way around.

File: src/parseSerialize/schemas.ts [hunk 3]
```diff
@@ -347,21 +343,20 @@ export const transferSchemaV1: TSchema = {
...
     txFields.fee,
-    txFields.recipient,
-    txFields.attachment,
   ],
 }
 
-export const sponsorSchemaV1: TSchema = {
+export const cancelSponsorSchemaV1: TSchema = {
   type: 'object',
   schema: [
```
Comment (KaloyanTanev) [inline src/parseSerialize/schemas.ts:361]: Same thing as above, for the order of timestamp and fee.

File: src/parseSerialize/schemas.ts [hunk 4]
```diff
@@ -374,29 +369,51 @@ export const sponsorSchemaV1: TSchema = {
...
+    txFields.version,
+    txFields.type,
+    txFields.senderPublicKey,
+    txFields.recipient,
+    txFields.amount,
     txFields.fee,
+    txFields.timestamp,
   ],
 }
+// ************************************************************************
```
Comment (KaloyanTanev) [inline src/parseSerialize/schemas.ts:404]: Funky line :)

File: src/parseSerialize/schemas.ts [hunk 5]
```diff
@@ -374,29 +369,51 @@ export const sponsorSchemaV1: TSchema = {
...
 
 /**
  * Maps transaction types to schemas object. Schemas are written by keys. 0 - no version, n - version n
  */
 export const schemasByTypeMap = {
   [TRANSACTION_TYPE.GENESIS]: {},
   [TRANSACTION_TYPE.TRANSFER]: {
-    1: transferSchemaV1,
```
Comment (KaloyanTanev) [inline src/parseSerialize/schemas.ts:396]: Shouldn't we still have transferV1? So it is:
- `transferSchemaV1`
- `transferSchemaV2`
- `transferSchemaV3` (later on)
- `transferSchemaLedger`

**Review by KaloyanTanev — CHANGES_REQUESTED** [review]

File: src/parseSerialize/schemas.ts (hunk 1 above)
Comment (brunodb3) [reply src/parseSerialize/schemas.ts:334]: That is added on the serializer for every array field :)

https://github.com/ltonetwork/lto-transactions/blob/1517c5916cccbed8857165733166c002f1bcf466/src/parseSerialize/serialize.ts#L23

File: src/parseSerialize/schemas.ts (hunk 4 above)
Comment (brunodb3) [reply src/parseSerialize/schemas.ts]: hehe removed it :)

File: src/parseSerialize/schemas.ts (hunk 5 above)
Comment (brunodb3) [reply src/parseSerialize/schemas.ts]: I renamed the schemas according to what Arnold mentioned on Telegram. Now we have `transferSchemaV1` and `transferSchemaV2`

**Review by KaloyanTanev — APPROVED** [review]

**Pull request merged**